src/Features/Core/Portable/ChangeSignature/AbstractChangeSignatureService.cs

```diff
@@ -899,6 +923,44 @@ protected static int GetParameterIndex<TNode>(SeparatedSyntaxList<TNode> paramet
             return Generator.SeparatedList(fullList.ToImmutableAndFree(), separators.ToImmutableAndFree());
         }
 
+        private async Task<(SyntaxNode expression, bool callsiteInferenceFailed)> GenerateInferredExpressionAsync(
+            SemanticModel semanticModel,
+            Workspace workspace,
+            int position,
+            ITypeSymbol targetType,
+            CancellationToken cancellationToken)
+        {
+            var recommendations = await Recommender.GetImmutableRecommendedSymbolsAtPositionAsync(
+                semanticModel, position, workspace, cancellationToken: cancellationToken).ConfigureAwait(false);
+
+            var sourceSymbols = recommendations.Where(r => r.IsNonImplicitAndFromSource());
+            var orderedLocalAndParameterSymbols = sourceSymbols
+                .Where(s => s.IsKind(SymbolKind.Local) || s.IsKind(SymbolKind.Parameter))
+                .OrderByDescending(s => s.Locations.First().SourceSpan.Start);
+
+            var orderedPropertiesAndFields = sourceSymbols
+                .Where(s => s.IsKind(SymbolKind.Property) || s.IsKind(SymbolKind.Field))
+                .OrderByDescending(s => s.Locations.First().SourceSpan.Start);
+
+            var fullyOrderedSymbols = orderedLocalAndParameterSymbols.Concat(orderedPropertiesAndFields);
+
+            foreach (var symbol in fullyOrderedSymbols)
+            {
+                var symbolType = symbol.GetSymbolType();
+                if (symbolType == null)
+                {
+                    continue;
+                }
+
+                if (semanticModel.Compilation.ClassifyCommonConversion(symbolType, targetType).IsImplicit)
+                {
+                    return (Generator.IdentifierName(symbol.Name), callsiteInferenceFailed: false);
+                }
+            }
+
+            return (expression: null, callsiteInferenceFailed: true);
```

Comment:
`callsiteInferenceFailed` seems reundant.  it seems to just be equivalent to `expression != null`.

Reply:
It had one subtlety that I've now done away with, and simplified the return type to just Expression.